src/goto-instrument/contracts/utils.cpp

```diff
@@ -74,6 +59,35 @@ void havoc_if_validt::append_scalar_havoc_code_for_expr(
   });
 }
 
+void havoc_assigns_targetst::append_havoc_code_for_expr(
+  const source_locationt location,
+  const exprt &expr,
+  goto_programt &dest) const
+{
+  if(expr.id() == ID_pointer_object)
+  {
+    append_object_havoc_code_for_expr(location, expr.operands().front(), dest);
+    return;
+  }
+
+  havoc_utilst::append_havoc_code_for_expr(location, expr, dest);
+}
+
+goto_programt::instructiont &
+add_pragma_disable_assigns_check(goto_programt::instructiont &instr)
+{
+  instr.source_location_nonconst().add_pragma(
+    CONTRACT_PRAGMA_DISABLE_ASSIGNS_CHECK);
+  return instr;
+}
+
+goto_programt &add_pragma_disable_assigns_check(goto_programt &prog)
+{
+  Forall_goto_program_instructions(it, prog)
+    add_pragma_disable_assigns_check(*it);
+  return prog;
+}
+
```

Comment:
Am I missing something or have these two functions (this one + the one above) just been moved to a different location in the same file? If so, can we move those back to reduce the diff?